test/vendor/github.com/Microsoft/hcsshim/ext4/tar2ext4/tar2ext4.go

```diff
@@ -268,3 +263,37 @@ func ReadExt4SuperBlock(vhdPath string) (*format.SuperBlock, error) {
 	}
 	return &sb, nil
 }
+
+// ConvertAndComputeRootDigest writes a compact ext4 file system image that contains the files in the
+// input tar stream, computes and returns its cryptographic digest. Convert is called with minimal
+// options: ConvertWhiteout and MaximumDiskSize set to dmverity.RecommendedVHDSizeGB.
+func ConvertAndComputeRootDigest(r io.Reader) (string, error) {
+	out, err := ioutil.TempFile("", "")
```

Comment:
A function that converts a tar stream and returns its Merkel tree should be useable by both Convert with append verify and by the policy code.

Reply:
done. Let me know if this is what was in your mind.